superset-frontend/src/views/chartList/ChartList.tsx

```diff
@@ -311,11 +320,24 @@ class ChartList extends React.PureComponent<Props, State> {
       },
       loading: true,
     });
-    const filterExps = filters.map(({ id: col, operator: opr, value }) => ({
-      col,
-      opr,
-      value,
-    }));
+    const filterExps = filters
+      .map(({ id: col, operator: opr, value }) => ({
+        col,
+        opr,
+        value,
+      }))
+      .reduce((acc, fltr) => {
+        if (fltr.col === 'datasource' && typeof fltr.value === 'object') {
```

Comment:
checking `fltr. value === 'object'` will be true if the value is null, is that intentional?

Reply:
there's another null check bellow. I'll move it up here so it's more visible